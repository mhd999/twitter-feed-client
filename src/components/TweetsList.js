// @flow

import React, { Component } from 'react';
import Tweet from './Tweet';
import { API_URL } from '../config';

const es = new EventSource(`${API_URL}/dashboard/feed`);

class TweetsList extends Component {
    constructor(props: any) {
        super(props);
        this.state = {
            tweets: [],
        };
    }
    state : {
        tweets: any[]
    };
    
    componentDidMount() {
        let tweets = this.props.tweets;
        es.onmessage =  event => {
            tweets.push(event.data);
            this.setState({ tweets: tweets });
        } 
    }

    render () {
        let tweetList = [];
        
        tweetList = this.props.tweets.map((tweet, index) => (
            <Tweet key={index} data={JSON.parse(tweet)} />  
        ));
        return(
            <ul>
                {tweetList}
            </ul>
        );
    }
}

export default TweetsList;
