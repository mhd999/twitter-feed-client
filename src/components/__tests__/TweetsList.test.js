import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TweetsList from '../TweetsList';

const tweets =[{"created_at":"Mon Dec 12 21:23:22 +0000 2016","id":808422013928935400,"text":"RT @wwf_uk: 70% of Norway’s wolf population could be hunted &amp; killed this winter. Help @WWFNorge #SaveOurWolves:… "}]

describe('<TweetsList />', () => {
  it('renders children', () => {
    const component = shallow(
      <TweetsList tweets={tweets} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
});
