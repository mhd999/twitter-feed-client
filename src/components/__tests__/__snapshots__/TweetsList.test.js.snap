exports[`<TweetsList /> renders children 1`] = `
<ul>
  <Component
    data={
      Object {
        "created_at": "Mon Dec 12 21:23:22 +0000 2016",
        "id": 808422013928935400,
        "text": "RT @wwf_uk: 70% of Norway’s wolf population could be hunted &amp; killed this winter. Help @WWFNorge #SaveOurWolves:… ",
      }
    } />
</ul>
`;
